### PR TITLE
Exception when email template parameter is an entity with non-UUID id entifier jmix-framework/jmix#1439

### PR DESCRIPTION
**Pull request by KremnevDmitry**

### CODE REVIEW

File: jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java [hunk 1]
```diff
@@ -66,12 +68,28 @@ public Object convertFromString(Class parameterClass, @Nullable String paramValu
...
-            UUID id = UUID.fromString(paramValueStr);
-            Optional entityOpt = dataManager.load(parameterClass)
-                    .id(id)
-                    .fetchPlan(FetchPlan.BASE)
-                    .optional();
-            return entityOpt.isPresent() ? entityOpt.get() : null;
+            MetaProperty idProperty = metadataTools.getPrimaryKeyProperty(parameterClass);
+            Object idValue = null;
+
+            if (idProperty != null && idProperty.getRange().isDatatype()) {
```
Comment (gorbunkov) [inline jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java:74]: Please check what happens if primary key is not a Datatype, e.g. the entity has a composite PK. It seems that the method will just return null. We should either handle composite primary keys or raise an exception saying that primary key type is not supported.

**Review by gorbunkov — CHANGES_REQUESTED** [review]

File: jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java (hunk 1 above)
Comment (KremnevDmitry) [reply jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java]: Now an exception will be thrown.

File: jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java [hunk 2]
```diff
@@ -66,12 +68,36 @@ public Object convertFromString(Class parameterClass, @Nullable String paramValu
...
+                if (idValue != null) {
+                    return dataManager.load(parameterClass)
+                            .id(Id.of(idValue, parameterClass))
+                            .fetchPlan(FetchPlan.BASE)
+                            .optional()
+                            .orElse(null);
+                }
+            } else if (idProperty.getRange().isClass()) {
+                throw new ReportingException(
+                        String.format("Unsupported composite id from [%s] with value [%s]",
```
Comment (gorbunkov) [inline jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java:96]: Let's rephrase: "composite id from" -> "composite primary key in"

**Review by gorbunkov — APPROVED** [review]

File: jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java (hunk 2 above)
Comment (KremnevDmitry) [reply jmix-reports/reports/src/main/java/io/jmix/reports/libintegration/JmixObjectToStringConverter.java]: Fixed

**Pull request merged**